daemon/api_users_test.go

```diff
@@ -837,6 +838,47 @@ func (s *userSuite) TestPostCreateUserFromAssertionAllKnownButOwned(c *check.C)
 	c.Check(rsp.Result, check.DeepEquals, expected)
 }
 
+func (s *userSuite) TestPostCreateUserAutomaticDisabled(c *check.C) {
+	s.makeSystemUsers(c, []map[string]interface{}{goodUser})
+
+	// disable automatic user creation
+	st := s.d.Overlord().State()
+	st.Lock()
+	tr := config.NewTransaction(st)
+	err := tr.Set("core", "users.create.automatic", false)
+	tr.Commit()
+	st.Unlock()
+	c.Assert(err, check.IsNil)
+
+	defer daemon.MockOsutilAddUser(func(username string, opts *osutil.AddUserOptions) error {
+		panic("no user should be created")
```

Comment:
nitpick^2: this could just set a bool flag and assert it in the test below, IMHO this is much nicer to read than panic if it fails (but really, it's a nitpick).

Reply:
I think the panics read OK,  and it's independent of the rest of the test, how nicely the error reads depends on details of the checks we do:

PANIC: api_users_test.go:841: userSuite.TestPostCreateUserAutomaticDisabled

... Panic: no user should be created (PC=0x431091)

/usr/lib/go-1.13/src/runtime/panic.go:679
  in gopanic
api_users_test.go:854
...